EIPS/eip-3171.md

```diff
@@ -0,0 +1,148 @@
+
+The preimage for the final hashing round is thus always exactly `53` bytes long (does not include hash of the init code).
+
+## Rationale
+
+### Address formula
+
+* As with CREATE2, ensures that addresses created with this scheme cannot collide with addresses created using the traditional
+  `keccak256(rlp([sender, nonce]))` formula, as `0xff` can only be a starting byte for RLP for data many petabytes long.
+* The addresses created with this scheme cannot collide with addresses created using the `CREATE2`, since the lengths 
+  of the preimage differ
+* Ensures that the hash preimage has a fixed size.
+
+### Gas cost
+
+The concerns for `CREATE2` gas cost in regard to init code preimage length do not apply, as the init code hash is not used
+in computing the account address.
+
+## Test Cases
+
+The addresses computed using the following example typescript code:
+
+```typescript
+import {utils} from 'ethers'
+
+function getCreate3Address(address: string, salt: string) {
+  const create3Inputs = ['0xff', address, salt]
+  // remove the 0x prefixes
+  const sanitizedInputs = `0x${create3Inputs.map((i) => i.slice(2)).join('')}`
+  return utils.getAddress(`0x${utils.keccak256(sanitizedInputs).slice(-40)}`)
+}
+```
+
+Example 0
+* address `0x0000000000000000000000000000000000000000`
+* salt `0x0000000000000000000000000000000000000000000000000000000000000000`
+* gas (assuming no mem expansion): *TBD*
+* result: `0x69aC1DB8a84821a964E54c0FE42E0E7C3B9b87D6`
+
+Example 1
+* address `0xdeadbeef00000000000000000000000000000000`
+* salt `0x0000000000000000000000000000000000000000000000000000000000000000`
+* gas (assuming no mem expansion): *TBD*
+* result: `0x2D758dBc3E8C5476fE9a3CB92D1594ea7Efbe7A2`
+
+Example 2
+* address `0xdeadbeef00000000000000000000000000000000`
+* salt `0x000000000000000000000000feed000000000000000000000000000000000000`
+* gas (assuming no mem expansion): *TBD*
+* result: `0x44339B41C67b140523a407dBf60b8394d9f3A67C`
+
+Example 3
+* address `0x00000000000000000000000000000000deadbeef`
+* salt `0x00000000000000000000000000000000000000000000000000000000cafebabe`
+* gas (assuming no mem expansion): *TBD*
+* result: `0x53647a8Cc98B0f82e2D1d27c8352025d09640b81`
+
+Example 4
+* address `0x00000000000000000000000000000000deadbeef`
+* salt `0x00000000000000000000000000000000000000000000000000000000cafebabe`
+* gas (assuming no mem expansion): *TBD*
+* result: `0x53647a8Cc98B0f82e2D1d27c8352025d09640b81`
+
+## Backwards Compatibility
+
+As in `CREATE2`, `CREATE3` makes collisions possible.
+
+`CREATE3` is not expected to have address collisions with `CREATE2`, as the preimage length differs by the `32` bytes
+of the init code hash.
+
+TODO: are there any existing protections in place that prevent contracts from self destructing and being recreated with `CREATE2`?
+
+## Security Considerations
+
+This new opcode allows the deterministic contract address to be independent of the bytecode. Users may not depend on the deterministic
```

Comment:
I'm not sure it's fair to say this opcode "allows" this behaviour as it's possible to deploy arbitrary contracts to deterministic addresses with `CREATE2`.